src/charm.py

```diff
@@ -33,9 +50,23 @@ def __init__(self, *args):
         self.framework.observe(
             self.on.spark_history_server_pebble_ready, self._on_spark_history_server_pebble_ready
         )
-        self.framework.observe(self.on.config_changed, self._on_config_changed)
         self.framework.observe(self.on.install, self._on_install)
-        self.spark_config = SparkHistoryServerConfig(self.model.config)
+        self.s3_creds_client = S3Requirer(self, S3_INTEGRATOR_REL)
+        self.framework.observe(
+            self.s3_creds_client.on.credentials_changed, self._on_s3_credential_changed
+        )
+        self.framework.observe(
+            self.on[S3_INTEGRATOR_REL].relation_joined, self._on_s3_credential_relation_joined
+        )
+        self.framework.observe(
```

Comment:
**issue**

There is already an event for this, I believe it is the CredentialsGone event which we should use. I don't think we should re-define the broken event, overriding the one in S3Requirer

@welpaolo ?

Reply:
Done, using CredentialsGone now.